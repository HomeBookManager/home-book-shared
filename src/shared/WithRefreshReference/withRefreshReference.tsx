import { FC } from 'react';

// utils
import { generateID } from '../../utils/common/generateID';

export const withRefreshReference: <T>(
  Component: FC<T>,
) => (props: T) => JSX.Element =
  <T,>(Icon: FC<T>) =>
  (props: T) => <Icon key={generateID()} {...props} />;

export default withRefreshReference;
