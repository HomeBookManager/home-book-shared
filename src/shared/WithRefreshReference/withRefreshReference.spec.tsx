import { render } from '@testing-library/react';

// components
import withRefreshReference from './withRefreshReference';

describe('withRefreshReference snapshots', () => {
  it('should render component with refresh reference', () => {
    // mock
    const Component = withRefreshReference(() => <></>);

    //before
    const { asFragment } = render(<Component />);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
