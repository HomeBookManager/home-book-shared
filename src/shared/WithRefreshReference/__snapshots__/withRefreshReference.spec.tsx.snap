// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRefreshReference snapshots should render component with refresh reference 1`] = `<DocumentFragment />`;
